src/metabase/models/setting.clj

```diff
@@ -370,11 +370,14 @@
 
 (defn- user-facing-info [setting]
   (let [k (:name setting)
-        v (get k)]
+        v (get k)
+        set-by-env (not (nil? (env-var-value setting)))]
     {:key         k
      :value       (when (and (not= v (env-var-value setting))
                              (not= v (:default setting)))
                     v)
```

Comment:
env var values are already hidden. See this code here.

I believe they are also "read-only" in the sense that the take precedent over anything set in the admin panel but you'll have to double-check that